src/plugins/friendsSince/index.tsx

```diff
@@ -4,50 +4,64 @@
+import { findByCodeLazy, findByPropsLazy, findLazy } from "@webpack";
+import { Heading, RelationshipStore, Text } from "@webpack/common";
 
-const container = findByPropsLazy("memberSinceWrapper");
+const containerWrapper = findByPropsLazy("memberSinceWrapper");
+const container = findByPropsLazy("memberSince");
 const getCreatedAtDate = findByCodeLazy('month:"short",day:"numeric"');
 const locale = findByPropsLazy("getLocale");
 const lastSection = findByPropsLazy("lastSection");
-
-const cl = classNameFactory("vc-friendssince-");
+const section = findLazy((m: any) => m.section !== void 0 && Object.values(m).length === 1);
 
 export default definePlugin({
     name: "FriendsSince",
     description: "Shows when you became friends with someone in the user popout",
-    authors: [Devs.Elvyra],
+    authors: [Devs.Elvyra, Devs.Antti],
     patches: [
-        // User popup
+        // User popup - old layout
         {
             find: ".USER_PROFILE}};return",
             replacement: {
                 match: /,{userId:(\i.id).{0,30}}\)/,
-                replace: "$&,$self.friendsSince({ userId: $1 })"
+                replace: "$&,$self.friendsSinceOld({ userId: $1 })"
             }
         },
-        // User DMs "User Profile" popup in the right
+        // DM User Sidebar - old layout
         {
             find: ".PROFILE_PANEL,",
             replacement: {
                 match: /,{userId:([^,]+?)}\)/,
-                replace: "$&,$self.friendsSince({ userId: $1 })"
+                replace: "$&,$self.friendsSinceOld({ userId: $1 })"
             }
         },
-        // User Profile Modal
+        // User Profile Modal - old layout
         {
             find: ".userInfoSectionHeader,",
             replacement: {
                 match: /(\.Messages\.USER_PROFILE_MEMBER_SINCE.+?userId:(.+?),textClassName:)(\i\.userInfoText)}\)/,
-                replace: (_, rest, userId, textClassName) => `${rest}!$self.getFriendSince(${userId}) ? ${textClassName} : void 0 }), $self.friendsSince({ userId: ${userId}, textClassName: ${textClassName} })`
+                replace: (_, rest, userId, textClassName) => `${rest}!$self.getFriendSince(${userId}) ? ${textClassName} : void 0 }), $self.friendsSinceOld({ userId: ${userId}, textClassName: ${textClassName} })`
+            }
+        },
+        // DM User Sidebar - new layout
+        {
+            find: ".PANEL}),nicknameIcons",
+            replacement: {
+                match: /USER_PROFILE_MEMBER_SINCE,.{0,100}userId:(.{0,3}\.id)}\)}\)/,
+                replace: "$&,$self.friendsSinceNew({userId:$1,isSidebar:true})"
+            }
+        },
+        // User Profile Modal - new layout
+        {
+            find: "action:\"PRESS_APP_CONNECTION\"",
+            replacement: {
+                match: /USER_PROFILE_MEMBER_SINCE,.{0,100}userId:(.{0,3}\.id),.{0,100}}\)}\),/,
+                replace: "$&,$self.friendsSinceNew({userId:$1,isSidebar:false}),"
             }
         }
```

Comment:
ah, so these were actually meant to match variable names. you can achieve that using our special `\i` escape
```suggestion
        {
            find: ".PANEL}),nicknameIcons",
            replacement: {
                match: /USER_PROFILE_MEMBER_SINCE,.{0,100}userId:(\i\.id)}\)}\)/,
                replace: "$&,$self.friendsSinceNew({userId:$1,isSidebar:true})"
            }
        },
        // User Profile Modal - new layout
        {
            find: "action:\"PRESS_APP_CONNECTION\"",
            replacement: {
                match: /USER_PROFILE_MEMBER_SINCE,.{0,100}userId:(\i\.id),.{0,100}}\)}\),/,
                replace: "$&,$self.friendsSinceNew({userId:$1,isSidebar:false}),"
            }
        }
```

Reply:
oh, cool